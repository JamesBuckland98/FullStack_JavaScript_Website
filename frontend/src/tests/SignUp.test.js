import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import SingUp from '../components/SignUp';

it('should render Sign up component correctly', () =>{
	const wrapper = shallow(<SingUp/>);
	expect(toJson(wrapper)).toMatchSnapshot();
});

describe('When sign up button is clicked', () => {
	it('switches between login and sign up', () => {
		const switchComponet = jest.fn();
		const wrapper = shallow(<SingUp switch= { switchComponet }/>);
		const buttonElement = wrapper.find('.link')
		buttonElement.simulate('click');
		expect(switchComponet).toHaveBeenCalledTimes(2);
	})
})