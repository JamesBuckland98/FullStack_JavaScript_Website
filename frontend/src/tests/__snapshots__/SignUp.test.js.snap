// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sign up component correctly 1`] = `
<div>
  <p>
     
     
  </p>
  <h3>
     Sign up: 
  </h3>
  <Formik
    initialValues={
      Object {
        "password": "",
        "username": "",
      }
    }
    onSubmit={[Function]}
    validate={[Function]}
  >
    <Component />
  </Formik>
  <button
    className="link"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
